src/vse_sync_pp/analyzers/analyzer.py

```diff
@@ -265,3 +270,193 @@ def explain(self, data):
             'duration': data.iloc[-1].timestamp - data.iloc[0].timestamp,
             'terror': self._statistics(data.terror, 'ns'),
         }
+
+
+class TimeIntervalErrorAnalyzerBase(Analyzer):
+    """Analyze Time Interval Error (also referred to as Wander).
+
+    Derived classes calculate specific Time Interval Error metric focused on measuring
+    the change of Time Error.
+    """
+    locked = frozenset()
+
+    def __init__(self, config):
+        super().__init__(config)
+        # samples in the initial transient period are ignored
+        self._transient = config.parameter('transient-period/s')
+        # minimum test duration for a valid test
+        self._duration = config.parameter('min-test-duration/s')
+        # limit initial tau observation windows from 1 to 9k taus
+        taus_below_tenkey = np.array([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 20, 30, 40, 50, 60, 70, 80, 90,
+                                     100, 200, 300, 400, 500, 600, 700, 800, 900, 1000, 2000, 3000,
+                                     4000, 5000, 6000, 7000, 8000, 10000])
+        # observation window upper bound limited to 100k samples in 5k increments
+        taus_above_tenkey = np.arange(10001, 100000, 5000)
+        # `taus_list` contains range limit of obervation window intervals for which to compute the statistic
+        self._taus_list = np.concatenate((taus_below_tenkey, taus_above_tenkey))
+        self._rate = None
+        self._lpf_signal = None
+        # atributtes to be initialized in derived class
+        self._accuracy = None
+
+    def calculate_limit(self, tau):
+        # return limit based on `tau`
+        if self._accuracy is None:
+            raise AttributeError('atributte not initialized')
```

Comment:
I don't think this is the right approach. Either `_accuracy` must be initialised in this class or `calculate_limit` lives elsewhere.

Reply:
One approach may be to make `calculate_limit` a regular function (at this module level?) ... `accuracy` would then passed as a mandatory argument. It will also allow `calculate_limit` to be isolated for unit testing.